### PR TITLE
[Core] Refactor core fixtures, add specific products types fixtures

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | yes
| Related tickets | #5285
| License         | MIT

 - [x] Extract `ExampleFactoryInterface` for resources
 - [x] Refactor and simplify `AbstractResourceFixture`
 - [x] Add helper for `OptionsResolver` default / normalizer closures
 - [x] **Specific products types:**
  - [x] Mugs
  - [x] Stickers
  - [x] T-shirts
  - [x] Books

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+                ->setNormalizer('shipping_category', LazyOption::findOneBy($shippingCategoryRepository, 'code'))
+                ->setDefault('taxons', LazyOption::randomOnes($taxonRepository, 3))
+                ->setAllowedTypes('taxons', 'array')
+                ->setNormalizer('taxons', LazyOption::findBy($taxonRepository, 'code'))
+                ->setDefault('channels', LazyOption::randomOnes($channelRepository, 3))
+                ->setAllowedTypes('channels', 'array')
+                ->setNormalizer('channels', LazyOption::findBy($channelRepository, 'code'))
+                ->setDefault('product_attributes', [])
+                ->setAllowedTypes('product_attributes', 'array')
+                ->setNormalizer('product_attributes', function (Options $options, array $productAttributesValues) use ($productAttibuteValueFactory) {
```
Comment (okwinza) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php]: i'd say we need some kind of ordering/sorting for all these calls so it'd be easier to read. WDYT?

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php]: The product example factory is the most complex one, but certainly we will, just not sure what kind :+1:

We can separate options from the other ones like:
```php
  ->setDefault('foo', ...)
  ->setAllowedTypes('foo', ...)
  ->setNormalizer('foo', ...)

  ->setDefault('bar', ...)
  // ...
```

or indent all option calls except the first one:

```php
  ->setDefault('foo', ...)
    ->setAllowedTypes('foo', ...)
    ->setNormalizer('foo', ...)
  ->setDefault('bar', ...)
  // ...
```

Comment (okwinza) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php]: or we can combine both :)

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    return StringInflector::nameToCode($options['name']);
+                })
+                ->setDefault('gateway', 'offline')
+                ->setDefault('enabled', function (Options $options) {
+                    return $this->faker->boolean(90);
+                })
+                ->setAllowedTypes('enabled', 'bool')
+        ;
+    }
+    /**
```
Comment (Arminek) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php]: Missing blank line 😱 .

Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php]: 😱

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php]: :scream:

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            $paymentMethod->setName(sprintf('[%s] %s', $localeCode, $options['name']));
+        }
+
+        return $paymentMethod;
+    }
+
+    /**
+     * @return array
+     */
+    private function getLocales()
```
Comment (Arminek) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php]: ``getLocaleCodes()`` WDYT?

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    {
+        /** @var LocaleInterface[] $locales */
+        $locales = $this->localeRepository->findAll();
+        foreach ($locales as $locale) {
+            yield $locale->getCode();
```
Comment (Arminek) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php]: 👍

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+            yield $locale->getCode();
+        }
+    }
+
+    /**
+     * @param AttributeInterface $productAttribute
+     *
+     * @return mixed
+     */
+    private function getRandomValueForProductAttribute(AttributeInterface $productAttribute)
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php]: 👍

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php (hunk 3 above)
Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/PaymentMethodExampleFactory.php]: In separate PR, pls! :)

**Comment by pjedrzejewski** [comment]
Thanks Kamil, much nicer than protected methods. ;)

**Pull request merged**